griptape/tasks/base_task.py

```diff
@@ -181,7 +185,18 @@ def after_run(self) -> None:
             )
 
     def can_run(self) -> bool:
-        return self.state == BaseTask.State.PENDING and all(parent.is_finished() for parent in self.parents)
+        # If this Task has been skipped or is not pending, it should not run
+        if self.is_skipped() or not self.is_pending():
+            return False
+
+        # If this Task has parents, and _all_ of them are skipped, it should not run
+        if self.parents and all(parent.is_skipped() for parent in self.parents):
+            return False
+
+        # If _all_ this Task's unskipped parents are finished, it should run
+        unskipped_parents = [parent for parent in self.parents if not parent.is_skipped()]
+
+        return all(parent.is_finished() for parent in unskipped_parents)
```

Comment:
I'm not sure this should be the default behavior. A Task may rely on the output of a parent and not know that it was skipped.

Airflow defaults to [all_success](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#trigger-rules) which I think we should follow.

What do you think @vachillo?

```suggestion
    def can_run(self) -> bool:
        # If this Task has been skipped or is not pending, it should not run
        if self.is_skipped() or not self.is_pending():
            return False

        # If this Task has parents, and _any_ of them are skipped, it should not run
        if self.parents and any(parent.is_skipped() for parent in self.parents):
            self.state = BaseTask.State.SKIPPED
            return False

        return all(parent.is_finished() for parent in self.parents)
```